content/posts/git-worktree-workflow/index.md

```diff
@@ -0,0 +1,455 @@
+unlock(ok): 05_visualization/240227J_Sim-Cre/displots/cell_03.png (file)        
+unlock(ok): 05_visualization/240227J_Sim-Cre/displots/cell_04.png (file)        
+unlock(ok): 05_visualization/240227J_Sim-Cre/displots/cell_06.png (file)        
+unlock(ok): 05_visualization/240227J_Sim-Cre/displots/cell_07.png (file)        
+unlock(ok): 05_visualization/240227J_Sim-Cre/displots/cell_08.png (file)        
+unlock(ok): 05_visualization/240227J_Sim-Cre/displots/cell_somata.png (file)    
+unlock(ok): 05_visualization/240227J_Sim-Cre/displots/cell_tuft.png (file)      
+unlock(ok): 05_visualization/240227J_Sim-Cre/rois/all_cells.png (file)          
+  [1 similar message has been suppressed; disable with datalad.ui.suppress-simil[INFO   ] == Command start (output follows) =====                               
+Traceback (most recent call last):
+  File "/mnt/Data/L5b-runs/code/src/process2p/generate_plots.py", line 9, in <module>
+    from process2p.utils import *
+ModuleNotFoundError: No module named 'process2p'
+[INFO   ] == Command exit (modification check follows) ===== 
+  [18 similar messages have been suppressed; disable with datalad.ui.suppress-similar-results=off]
+run(error): /mnt/Data/L5b-runs (dataset) [python code/src/process2p/generate_plots...]
+add(ok): 05_visualization/240227J_Sim-Cre/displots/cell_01.png (file)           
+add(ok): 05_visualization/240227J_Sim-Cre/displots/cell_02.png (file)           
+add(ok): 05_visualization/240227J_Sim-Cre/displots/cell_03.png (file)           
+add(ok): 05_visualization/240227J_Sim-Cre/displots/cell_04.png (file)           
+add(ok): 05_visualization/240227J_Sim-Cre/displots/cell_06.png (file)           
+add(ok): 05_visualization/240227J_Sim-Cre/displots/cell_07.png (file)           
+add(ok): 05_visualization/240227J_Sim-Cre/displots/cell_08.png (file)           
+add(ok): 05_visualization/240227J_Sim-Cre/displots/cell_somata.png (file)       
+add(ok): 05_visualization/240227J_Sim-Cre/displots/cell_tuft.png (file)         
+add(ok): 05_visualization/240227J_Sim-Cre/rois/all_cells.png (file)             
+  [1 similar message has been suppressed; disable with datalad.ui.suppress-simil  [18 similar messages have been suppressed; disable with datalad.ui.suppress-similar-results=off]
+action summary:                                                                 
+  add (ok: 28)
+  get (notneeded: 3)
+  run (error: 1)
+  save (notneeded: 5)
+  unlock (ok: 28)
+```
+What happened? - Apparently, I've lost the '.venv' directory during the hard reset which causes "ModuleNotFoundError: No module named 'process2p'". määäh! This actually speaks for the use of a container. The problem with the container is that I have to rebuild it everytime I update my code ... annoying! I guess that's the trade-off between efficiency and reproducibility. To illustrate this 'highly complex' dilemma with Deepseek's smart-ass comment in a graph:
```

Comment:
technically that's not true: you can use container as your "environment" and run outside script in it. You can even bind-mount that script over some version of that script which you might have already inside the container. So you can kinda have the best of both worlds... might want to adjust "problem statement" here ;)

Reply:
I would need to reread above to give more conclusive answer, but overall, if you had managed to commit `.venv` and `reset --hard` away from that -- it would be gone. If it was untracked, it would remain untracked. If you want to clean up -- do `git clean -dfx` . In tandem with `git reset --hard` it is a "go nuclear" resetting to entirely clean state.